### ISSUE TITLE
Snippet is inserted at wrong indentation

### ISSUE THREAD

**Issue by ghid** [andreyorst/SimpleSnippets.vim#34]
Hi.

I've more a question than an issue...

If I insert a snippet, it won't be inserted at the current level of indentation. It is alway inserted at column one. Are there special settings required (regarding tabstop, shiftwidth and so on)? It seem to work in gifs within the documentation though.

Thx in advance and greetings.

**Comment by andreyorst** [comment]
Do you have `filetype plugin indent on` in you vimrc?

**Comment by andreyorst** [comment]
@ghid Seems like this is something with your particular config, because I can't reproduce it with minimal `vimrc`:
```vim
set nocompatible
filetype off
set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()
Plugin 'VundleVim/Vundle.vim'
Plugin 'andreyorst/SimpleSnippets.vim'
Plugin 'andreyorst/SimpleSnippets-snippets'
call vundle#end()
```
For me it works in vim and neovim, with this small configuration.

So I definitively need your `vimrc`, and info about your workspace.

**Comment by ghid** [comment]
@andreyorst thx for support and the hint. I've tried it also with an minimal vimrc and wasn't able to reproduce it any more; so I think you're right, to presume the root cause may be within my _regular_ vimrc. I'll investigate this further. If you could leave this issue open I'd document what I find out. I'd close the issue in a few days if I can't figure out what the problem is.

**Comment by andreyorst** [comment]
> If you could leave this issue open I'd document what I find out.

Sure! No problem.

>  I'd close the issue in a few days if I can't figure out what the problem is.

I'd like this issue to be solved, so if there any way of helping out i'm fully open for any questions.

**Comment by ghid** [comment]
### Example:

The 2nd `if` isn't indented correctly (after moving to normal mode and back to insert mode again):

![simplesnippets_issue34_1](https://user-images.githubusercontent.com/7606219/40648906-dd7776da-632f-11e8-931e-2853c9bc64ed.gif)

### Here's my _vimrc:
```
set nocompatible
filetype off
set number
set nobackup
set autoindent

set packpath=$VIM/vimfiles
packadd minpac
call minpac#init()
call minpac#add('andreyorst/SimpleSnippets.vim')

let g:SimpleSnippets_search_path = $VIM."/vimfiles/snippets/"
```

And the snippets I used:

**func** snippet:
```
${1:function}(${2:params})
{
	${0:;TODO Add code}
}
```

**if** snippet:
```
if (${1:true})
{
	${0:; expression}
}
```

@andreyorst Does this give you a clue?

**Comment by andreyorst** [comment]
What file-type is it? If there is no file-type defined, then your code indented properly. SimpleSnippets uses `=` command to indent lines, so if you indent it manually by executing `gg=G` it would lead to the same result. That's because Vim didn't found line ending (`;`) for current file-type so it assumes that indent level should be increased. However it doesn't indent your code  automatically when you pressed `o` as you may notice, which leads me thinking that there is no file-type in current example. I don't know in which language comments are defined by `;`. But in terms of Vim code indented properly. You can try `gg=G` and you will see the same result.

**Comment by andreyorst** [comment]
I've found that in Go lang comments are starting with semicolon. So if yo do `:setf go` and try to execute those snippets it should be indented properly.
![image](https://user-images.githubusercontent.com/19470159/40655601-8b184f16-634a-11e8-913f-3157859ae4c6.png)

**Comment by ghid** [comment]
The code is AutoHotkey. Which uses a semicolon (;) for one-line-comments.

**Comment by andreyorst** [comment]
does `:setf autohotkey` help?
![image](https://user-images.githubusercontent.com/19470159/40655727-f08f5dee-634a-11e8-81e7-64f118fbfc57.png)

**Comment by andreyorst** [comment]
for the example you provided, can you show the output of `:echo &ft`? I think that Vim didn't detected filetype.

**Comment by ghid** [comment]
I did set this ft=autothotkey already above. But now, if I remove the semicolons from the snippets, they are inserted with the correct indentation.

**Comment by andreyorst** [comment]
hm. I've tested this in vim and neovim and it behaves differently. In vim it doesn't indent properly, but in neovim it works fine. Strange. Seems configuration related still, because neovim turns on some features by default

**Comment by andreyorst** [comment]
Weird. Setting filetype to Go and then back to AutoHotKey fixes the indentation for `=` command.

**Comment by andreyorst** [comment]
I've found some code over the internet from some forum:
```autohotkey
While GetKeyState("CAPSLOCK", "P")
{
    vimize()
    if modal != 
		Tooltip, %context%: %num%, 60, 10
	else if num !=
		Tooltip, %num%, 60, 10
	else
		Tooltip
	SetTimer, vim, off
}
```
and tried to add some comments to it.
It indents fine. However if I manually type your snippets (without simple snippets) the code will be indented wrong

**Comment by ghid** [comment]
I digged a little deeper... First of all: Vim has no indent script for  filetype autohotkey out-of-the-box. The funny thing is that this hasn't bothered me so far. 😆
Maybe the default behaviour is to indent a new paragraph; but I haven't checked yet.

So, if you one opens an autohotkey file and changes the filetype, for example, to *go* then `:set indentexpr?` would state `indentexpr=GoIndent(v:lnum)`. By changing the filetype back to *autohotkey*, for which no indent script is available so far, `:set indentexpr?` will furthermore show `indentexpr=GoIndex(v:lnum)` and `=` will use the Go indentation, which is fairly similar to autohotkeys.

But I've found an autohotkey indent script on github: https://github.com/hnamikaw/vim-autohotkey
I'll experiment a little with that and document what I'm finding out.
